userspace/libsinsp/parsers.cpp

```diff
@@ -4490,6 +4493,44 @@ void sinsp_parser::parse_setsid_exit(sinsp_evt *evt)
 	}
 }
 
+void sinsp_parser::parse_prctl_exit(sinsp_evt *evt)
+{
+	sinsp_evt_param *parinfo;
+	int64_t retval;
+	sinsp_threadinfo *tinfo = evt->get_thread_info();
+	//
+	// Extract the return value
+	//
+	parinfo = evt->get_param(0);
+	retval = *(int64_t *)parinfo->m_val;
+	ASSERT(parinfo->m_len == sizeof(int64_t));
+
+	if(retval >= 0)
+	{
+		// If the option is PR_SET_NAME, change the name of
+		// this thread's program.
+		parinfo = evt->get_param(1);
+		int64_t option = *(int64_t *)parinfo->m_val;
+
+		if(option == PPM_PR_SET_NAME)
```

Comment:
have you checked if `prctl` apply only to the caller thread or to the whole process? In that case probably you should change all the `tinfo`s of that process?

Reply:
Yeah, prctl() only changes the thread's name:

```
#include <thread>
#include <sys/types.h>
#include <sys/prctl.h>
#include <stdio.h>

void new_thread()
{
        char buf[24];

        prctl(PR_SET_NAME, "my prog", NULL, NULL, NULL);
        prctl(PR_GET_NAME, buf, NULL, NULL, NULL);
        printf("Thread name: %s\n", buf);
}


int main(int argc, char **argv)
{
        char buf[24];

        std::thread t1(new_thread);
        t1.join();

        prctl(PR_GET_NAME, buf, NULL, NULL, NULL);
        printf("Main name: %s\n", buf);
}
```

```
$ ~/prctl_thread
Thread name: my prog
Main name: prctl_thread
```